pkg/ociinstaller/ocidownloader.go

```diff
@@ -27,27 +35,86 @@ func NewOciDownloader() *ociDownloader {
 	}
 }
 
-/* 
+/*
 Pull downloads the image from the given `ref` to the supplied `destDir`
 
 Returns
 
 	imageDescription, configDescription, config, imageLayers, error
 */
 func (o *ociDownloader) Pull(ctx context.Context, ref string, mediaTypes []string, destDir string) (*ocispec.Descriptor, *ocispec.Descriptor, []byte, []ocispec.Descriptor, error) {
-	log.Println("[TRACE] ociDownloader.Pull:", "pulling", ref)
-	fileStore := content.NewFileStore(destDir)
+	split := strings.Split(ref, ":")
+	tag := split[len(split)-1]
+	log.Println("[TRACE] ociDownloader.Pull:", "preparing to pull ref", ref, "tag", tag, "destDir", destDir)
+
+	// Create the target file store
+	memoryStore := memory.New()
+	fileStore, err := file.NewWithFallbackStorage(destDir, memoryStore)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
 	defer fileStore.Close()
 
-	hybridStore := newHybridStore(fileStore)
-	pullOpts := []oras.PullOpt{
-		oras.WithAllowedMediaTypes(append(mediaTypes, MediaTypeConfig, MediaTypePluginConfig)),
-		oras.WithPullEmptyNameAllowed(),
+	// Connect to the remote repository
+	repo, err := remote.NewRepository(ref)
+	if err != nil {
+		return nil, nil, nil, nil, err
 	}
-	desc, layers, err := oras.Pull(ctx, o.resolver, ref, hybridStore, pullOpts...)
+
+	// Get credentials from the docker credentials store
+	storeOpts := credentials.StoreOptions{}
+	credStore, err := credentials.NewStoreFromDocker(storeOpts)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	// Prepare the auth client for the registry and credential store
+	repo.Client = &auth.Client{
+		Client:     retry.DefaultClient,
+		Cache:      auth.DefaultCache,
+		Credential: credentials.Credential(credStore), // Use the credential store
+	}
+
+	// Copy from the remote repository to the file store
+	log.Println("[TRACE] ociDownloader.Pull:", "pulling...")
+
+	copyOpt := oras.DefaultCopyOptions
+	// TODO: use WithTargetPlatform to limit downloads
+	// copyOpt.WithTargetPlatform(&ocispec.Platform{
+	// 	Architecture: "amd64",
+	// 	OS:           "linux",
+	// })
+	manifestDescriptor, err := oras.Copy(ctx, repo, tag, fileStore, tag, copyOpt)
 	if err != nil {
-		return &desc, nil, nil, layers, err
+		log.Println("[ERROR] ociDownloader.Pull:", "failed to pull", ref, err)
```

Comment:
Make this a `TRACE` level log. Raw log messages should not end up in the console if we can avoid it.

Reply:
Done!